### PR TITLE
Add collation argument to the create Managed Instance command

### PR DESCRIPTION
**Pull request by NMijat1024**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/_params.py [hunk 1]
```diff
@@ -1073,6 +1073,10 @@ def _configure_security_policy_storage_params(arg_ctx):
                    options_list=['--capacity', '-c'],
                    help='The capacity of the managed instance in vcores.')
 
+        c.argument('collation',
+                   options_list=['--collation'],
```
Comment (adewaleo) [inline src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/_params.py]: Using `options_list` here is unnecessary. The cli option / parameter will be automatically built from the function dest parameter 'collation'. Use `options_list` when you want the cli parameter to have a different name or when you want to add a short parameter as in line 1074 above.

**Comment by tjprescott** [comment]
@NMijat1024 if you would like to get this into this release, please get the CI passing by this afternoon.

**Comment by williexu** [comment]
@NMijat1024 CI is now failing due to Pep8. In addition, I have submitted a fix for the issue we talked about: https://github.com/Azure/azure-cli/pull/8063
Once that is merged, you can re-record the test and update this PR.

**Comment by jaredmoo** [comment]
FYI style can be validated using this command: `azdev style --pep8 --pylint --module sql`

**Comment by jaredmoo** [comment]
@tjprescott , @williexu , we would like this change to make today's release train. We're happy to work with Willie's change, but this PR should not block on that one. A test refactoring should not block product release. Please merge

**Comment by tjprescott** [comment]
@NMijat1024 please resolve the conflict (likely, re-record the conflicting test) and push and we can get this into the hotfix release we are planning.

**Comment by jaredmoo** [comment]
I'll do this today since @NMijat1024 is in Serbia and is off work for the day.

**Comment by jaredmoo** [comment]
Fixed in branch https://github.com/jaredmoo/azure-cli/tree/collation-arg .

**Comment by tjprescott** [comment]
@jaredmoo can you open a PR against the release branch based on your branch?

**Comment by jaredmoo** [comment]
https://github.com/Azure/azure-cli/pull/8100

**Comment by tjprescott** [comment]
Thanks @jaredmoo. I will close this PR in favor of #8100.

**Pull request closed without merging**